hudi-common/src/main/java/org/apache/hudi/common/table/read/FileGroupReaderSchemaHandler.java

```diff
@@ -74,6 +74,10 @@ public class FileGroupReaderSchemaHandler<T> {
   // requiredSchema: the requestedSchema with any additional columns required for merging etc
   protected final Schema requiredSchema;
 
+  // the schema for incoming records, usually it equals with the requiredSchema,
+  // the only exception is for COW merge cases and MOR small file handling, which do not include the metadata fields.
+  protected Schema incomingRecordSchema;
```

Comment:
this could also be records from log files, maybe name it `incRecordSchema` or `deltaRecordSchema`?

Reply:
will go w/ `updatesRecordSchema`